src/awkward1/operations/convert.py

```diff
@@ -1139,6 +1139,180 @@ def getfunction(layout, depth):
     else:
         return out
 
+def to_arrow(layout):
+    import pyarrow
+
+    def recurse(layout, mask = None):
+
+        if isinstance(layout, awkward1.layout.NumpyArray):
+            numpy_arr = numpy.asarray(layout)
+            if(numpy_arr.ndim == 1):
+                return pyarrow.array(numpy_arr, mask=mask)
+            else:
+                return pyarrow.Tensor.from_numpy(numpy_arr)
+        
+        elif isinstance(layout, awkward1.layout.EmptyArray):
+            return pyarrow.array(numpy.array([], dtype=numpy.float64))
+
+        elif isinstance(layout, (awkward1.layout.IndexU8, 
+                                awkward1.layout.IndexU32, 
+                                awkward1.layout.Index8, 
+                                awkward1.layout.Index32, 
+                                awkward1.layout.Index64)):
+            return pyarrow.array(numpy.asarray(layout))
+        
+        elif isinstance(layout, awkward1.layout.ListOffsetArray32):
+
+            if (awkward1.operations.describe.parameters(layout).get("__array__") == "bytestring"):
+                arr = [awkward1.behaviors.string.ByteBehavior(
+                    layout[i]).__bytes__() for i in range(len(layout))]
+                if mask is not None:
+                    return pyarrow.array(arr, mask=numpy.array(mask, dtype=numpy.bool))
+                else:
+                    return pyarrow.array(arr)
+
+            if (awkward1.operations.describe.parameters(layout).get("__array__") == "string"):
+                arr = [awkward1.behaviors.string.CharBehavior(
+                    layout[i]).__str__() for i in range(len(layout))]
+                if mask is not None:
+                    return pyarrow.array(arr, mask=numpy.array(mask, dtype=numpy.bool))
+                else:
+                    return pyarrow.array(arr)
+            
+            if mask is not None:
+                indices = []
+                for i in range(len(mask)):
+                    if mask[i] == 1:
+                        indices.append(None)
+                    else:
+                        indices.append(i)
+                return pyarrow.ListArray.from_arrays(recurse(layout.offsets), 
+                                                            recurse(layout.content)).take( pyarrow.array(indices))
```

Comment:
Okay, this might seem a bit unorthodox in the sense that I am trying to get the Arrow Array to represent the Awkward Array as closely as possible since in a much broader sense `Arrow` is ultimately a means to `view` the `buffers` and `Awkward` is a `Data Manipulation Library`. I had two choices here, first to use the `filter` method which negates the null values and all you can see is the True values or second use `take` which makes up for the False values as null. I chose the second, because of the reasons I stated above. I'd like to have your view(@jpivarski ) on this.

Reply:
Building up `indicies` as a Python list is a problem.

Probably what we should do is schedule a pair-programming session. I've done that a few times with Atom TeleType with Skype/Zoom and it's worked very well.

Monday or Tuesday of next week? How late would be okay for you? (I'm not trying to avoid getting up early; I'm trying to avoid morning meetings because that's prime time for CERN-U.S. meetings. I'm in a project that has takes a lot of that time, but if you name an hour-long slot that works for you, I'll skip it to help you out here.)

Reply:
For now, I am making anything related to indices is go to a secondary function(helper), so that if there's any drastic change required based on our meeting tomorrow, it would be easier to deal with that in a single function. Just wanted to know, is the idea of `take` and `indices` is a problem or the implementation of extracting the `indices`.